torch_geometric/transforms/half_hop.py

```diff
@@ -0,0 +1,91 @@
+import torch
+
+from torch_geometric.data import Data
+from torch_geometric.data.datapipes import functional_transform
+from torch_geometric.transforms import BaseTransform
+
+
+@functional_transform('half_hop')
+class HalfHop(BaseTransform):
+    r"""Graph upsampling augmentation from the
+    `"Half-Hop: A graph upsampling approach for slowing down
+    message passing" <https://openreview.net/forum?id=lXczFIwQkv>`_ paper.
+    The graph is augmented by adding artificial slow nodes between neighbors
+    to slow down message propagation. (functional name: :obj:`half_hop`).
+
+    Args:
+        alpha (float, optional): The interpolation factor for
+            the slow nodes. (default: :obj:`0.5`)
+        p (float, optional): The probability of half-hopping
+            an edge. (default: :obj:`1.0`)
+
+    .. code-block:: python
+
+        import torch_geometric.transforms as T
+
+        data = ...
+        data = T.HalfHop(alpha=0.5)(data) # Apply transformation
+        out = model(data.x, data.edge_index) # Feed-forward
+        out = out[~data.slow_node_mask] # Get rid of slow nodes
+    """
+    def __init__(self, alpha: float = 0.5, p: float = 1.0):
+        if alpha < 0. or alpha > 1.:
+            raise ValueError(f'Interpolation factor has to be between 0 and 1 '
+                             f'(got {alpha}')
+        if p < 0. or p > 1.:
+            raise ValueError(
+                f'Ratio of half-hopped edges has to be between 0 and 1 '
+                f'(got {p}')
+        self.p = p
+        self.alpha = alpha
+
+    def forward(self, data: Data) -> Data:
+
+        assert data.edge_weight is None
+        assert data.edge_attr is None
```

Comment:
```suggestion
        if data.edge_weight is None or data.edge_attr is None:
            raise ValueError("'HalfHop augmentation is not support if "
                                         "'data' has 'edge_weight' or 'edge_attr'")
```

Reply:
Mention in the docs that `data` with `edge_weight` or `edge_attr` is not supported.